cocotb/handle.py

```diff
@@ -956,7 +955,7 @@ def SimHandle(handle, path=None):
         return obj
 
     if t not in _type2cls:
-        raise TestError("Couldn't find a matching object for GPI type %d (path=%s)" % (t, path))
+        raise TypeError("Couldn't find a matching object for GPI type %d (path=%s)" % (t, path))
```

Comment:
I would go for `RuntimeError` (this is a problem with cocotb) or `NotImplementedError` (please make a PR for this!).

TypeError is usually for "this argument is the wrong type", not "a type could not be found".

Reply:
```suggestion
        raise NotImplementedError("Couldn't find a matching object for GPI type %d (path=%s)" % (t, path))
```